internal/tests/serialization/set.go

```diff
@@ -0,0 +1,212 @@
+package serialization
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"reflect"
+	"runtime/debug"
+	"testing"
+)
+
+// Set is a tool for generating test cases of marshal and unmarshall funcs.
+// For cases when the function should no error,
+// marshaled data from Set.Values should be equal with Set.Data,
+// unmarshalled value from Set.Data should be equal with Set.Values.
+type Set struct {
+	Data   []byte
+	Values []interface{}
+
+	BrokenMarshalTypes   []reflect.Type
+	BrokenUnmarshalTypes []reflect.Type
+}
+
+func (s Set) RunTest(name string, t *testing.T, marshal func(interface{}) ([]byte, error), unmarshal func([]byte, interface{}) error) {
+	if name == "" {
```

Comment:
Since you introduced a new method, now you need to explain what is the purpose of `RunTest`, outlining difference between `RunCorruptTest`

Reply:
Done